### PR TITLE
Create My budget app

### PR DESCRIPTION
**Pull request by Sahar-AbdelSamad**
# Hi there 👋🏻 

Thanks for taking the time to review the project. Here are some things important to note about it:

### Changes implemented

- Created `splash` page that hash the name of the app, and links to the `sign up` and `log in` pages.
- Created `Sign up` and `log in` pages
- Created the `home` page (catagories page)
    - In the home page, when the user clicks (or taps) on a category item, the application navigates to the `transactions` page for that category.
    - The `home` page has a button `Add a new category` at the bottom that brings the user to the page to create a new category.

- Created `transactions` page
    - For a given category, the list of transactions is presented, ordered by the most recent.
    - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
    - There is a button `Add a new transaction` at the bottom that brings the user to the page to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

- Created `Add a new category` page
    - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
    - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

- Created `Add a new transaction` page
    - The user fills out a form to create a new transaction.
    - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- Created unit and integration tests for all the most important components of the application.
- Updated the README file
- Deployed the app

## Final notes

Is expected that this project meets the requirements. Please let us know if anything is left to be done or can be improved. Again, thanks for the review.

### CODE REVIEW

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<div>
+  <% @categories.each do |category| %>
+    <%= render category %>
+  <% end %>
+</div>
+
+<button class='add-category btn flex'>
+<div class='new-category btn flex'>
+  <%= link_to "Add a new Category", new_category_path %>
+  <div>
```
Comment (andersonlebon) [inline app/views/categories/index.html.erb:13]: - [ ] Kindly note that the `add new categories` and `new transaction` buttons are always displayed at the bottom of the view-heigh even though the list of all categories/transactions is empty. Check the deference here :point_down: 

your design | the original template
 -------|----------
image | image

- [ ] Kindly also remove the margin around your app as the original template does not have one.
- [ ] Make sure also you are using the same `font-family`

File: README.md [hunk 2]
```diff
@@ -1 +1,69 @@
...
+- _**PostgreSQL**_
+
+### Preview
+
+![This is gif of our project](screenshot.gif)
+
+## Live Demo
+
+[Live Demo Link](https://thawing-sierra-32352.herokuapp.com/)
+
```
Comment (andersonlebon) [inline README.md:20]: - [ ] Kindly make sure you follow the requirement for the project documentation. You need to record a video with :
- a description of the project.
- a demo of the project features.
- you should also highlight some interesting piece of code or something you built that you are very proud of.

image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Sahar-AbdelSamad 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required Changes ♻️

- [ ] Kindly check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (Sahar-AbdelSamad) [reply README.md:20]: Forgot to add it 🙈

File: app/views/categories/index.html.erb (hunk 1 above)
Comment (Sahar-AbdelSamad) [reply app/views/categories/index.html.erb:13]: Done 👍 thank you for pointing these out 😄

**Review by elmar8287 — APPROVED** [review]
Hi, 

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**